### ISSUE TITLE
CSS-issue with the striped table rows in /admin/users

### ISSUE THREAD

**Issue by Greenheart** [teknik-eksjo/chronos#56]
**Note:** I added this to the second milestone since this isn't part of the app's core functionality.
## 

Rows in the table are painted according to their css when then page initially renders, but when a user searches for users, the rows don't have adjusted colors so every second row get a darker color. Instead the rows keep their initial styling because it's applied to every `nth-child(2)` of the `tbody`.
## Possible solution:
- apply `:nth-child(2)` to make every second item have a color but also apply`:not('.hidden')` to make sure that the styling only **works with visible rows**

**Comment by shhmon** [comment]
Well as we saw, that solution didn' t quite work right now. I think i might have messed up with the CSS selectors tho, so we should try to fiddle with this some more. Why isn't anything from Hugos latest fork changes pull-requested to this repo yet?

**Comment by hugolundin** [comment]
My changes have been a part of the admin ui which I started working on in a seperate branch. Maybe I should send a pull-request for it, so we can work on smaller features of the admin ui instead. @Greenheart @Catagen @Limpan

**Comment by shhmon** [comment]
@hugolundin sounds like a plan. The table and search functionality especially seems to require contribution from everyone right now so it might be a good idea to go with smaller pull-requests so that everyone can help when it comes to admin UI features.

**Comment by Limpan** [comment]
Either send the pull request to develop or maybe we should have a new branch for it?

**Comment by shhmon** [comment]
@Limpan I guess that depends on if we're planning to start working on something else simultaneously during the development of the admin UI. I guess a new branch wouldn't hurt. Any thoughts, @hugolundin @Greenheart ?

**Comment by Limpan** [comment]
If @hugolundin pushes his branch to upstream you can all work on it and then we can clean up the commit history and merge into develop. I think it will be simpler to arrange that way.

**Comment by hugolundin** [comment]
Do I have permissions to push my branch to upstream?

**Comment by Greenheart** [comment]
I agree that we should open up feature/admin-ui to allow more of us to work on that part of the app.

It feels like @hugolundin and @Limpan  are very good at locking up pretty big parts of the app with each of his branches. Wouldn't it be better if we focus more on creating branches that fix **smaller** seperate things to ease collaboration and allow the whole team to contribute?

For example I think @Catagen @erikcollerstedt @Kallshem and I have felt locked out for the last couple of weeks... ;)

## Edit

At the same time, it is possible to fork the big fork like I did yesterday when submiting a PR to @hugolundin 's branch. But it's requiring extra work from the team

---

## A suggestion for how we should continue (Lists ftw... :smile:!)
1. Hugo shares his branch `feature/admin-ui` (But **where** and **how**?)
2. Everyone makes sure to tell the team what features or fixes they **are** or **will** be working on by assigning to issues
3. Make branches with the style `feature/admin-dashboard-chart-component` or `fix/users-list-col-width`. Make sure that the names describe what changes the branch will contain. Also pick smaller, specific names to not "lock" a bigger chunk of the app for example.
4. Make PR's to `teknik-eksjo/chronos/develop`, wait for QA and make improvements based on feedback or hav ethe changes merged

---

Let's go!

**Comment by hugolundin** [comment]
I think you wrote the exactly same thing as they have discussed above @Greenheart, but of course I agree with you 😋 The problem with the admin-ui branch is that it have gotten a bigger role in the project than I first expected - I just wanted to create a branch for integrating my testing of excel-uploads in the project, and we had not started to create issues/discuss features yet. Therefore I did not think about how I named my branch 😉 
It was never my intention to work on the whole admin-ui or exclude all of you, but experiencing this kind of things probably will help us work better together in the long run. 

If I have the right permissions to push my branch to upstream, I will do that asap 😊

**Comment by Greenheart** [comment]
> Therefore did not think about how I named my branch :wink:
> It was never my intention to work on the whole admin-ui and exclude you, but experiencing this kind of things probably will help us work better in the long run.

I agree, let's make use of this situation and improve our workflow! :smile: +1

---

@hugolundin I think we should clean up your branch's commit-history a bit (from merges / reverts) and then merge it back into `teknik-eksjo/chronos/develop`.
@Limpan Maybe a `git rebase` could work?

**Comment by hugolundin** [comment]
When merging a pull request, aren't one able to squash the commits?

**Comment by Greenheart** [comment]
### Edit: [This guide might be better](https://github.com/blog/2141-squash-your-commits)

@hugolundin Good idea! :smile: [This guide](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request) walks you through exactly how to do that.
1. Pull the latest remote version of `feature/admin-ui` to your computer
2. Perform some git magic locally as explained in the tutorial
3. Force-push back to `your-fork/admin-ui`
4. PR to `main-repo/develop`

Good luck! :)

**Comment by Limpan** [comment]
You should have permission to push.

Before pushing, go through the history and choose a commit that will be the limit for how far into the past you modify. `git log --oneline` works well for this.

Then run `git rebase -i XXXXXXX` to change the history. Choose what to do with each commit, modify commit messages so they look nice. (Choose a text editor before doing this since most people will be confused by the default vim.)

**Comment by hugolundin** [comment]
Why not use https://github.com/blog/2141-squash-your-commits?

**Comment by Limpan** [comment]
That will only give you one commit when done. When you rebase it is my understanding that you keep however many commits you want. It is also possible to do rebase many times to successively squash commits.

It's probably better to have a few commits rather than one massive commit.

https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

**Comment by hugolundin** [comment]
@Greenheart said that I should push my rebased version into my-fork/admin-ui and then send a pr to main-repo/develop. Why are we even talking about my permissions? 😅

**Comment by Greenheart** [comment]
@hugolundin When will you do the rebase? I'll wait with the admin-ui-related issues until the merge is completed :)

**Comment by Limpan** [comment]
@hugolundin The reason we got into a permissions discussion was that we talked about creating a new branch in teknik-eksjo/chronos.

**Comment by Greenheart** [comment]
Me and @hugolundin will fix the rebase either tomorrow evening or Sunday evening. So on Monday, the ones who want should be able to branch off in several smaller branches and continue the work again! :)

@teknik-eksjo/te13 I've updated the [Contributing Guidelines](https://github.com/teknik-eksjo/chronos/wiki/Contribution-Guidelines) to clarify how I suggest we work with branches etc. Feel free to give some feedback! :)

**Comment by shhmon** [comment]
Comming back to the actual issue - I've looked around and this striped-row bug seems to be a well-known bootstrap problem. There is no 'pretty' way to solve it from what I can see, so I'm thinking that maybe we should just skip the stripes instead of forcing it. The only solution I can come up with is ignoring the bootstrap styling and working with individual classes for every element of the table, which is kind of a pain.
Thoughts, @hugolundin @Greenheart ?

**Comment by Greenheart** [comment]
I think the interface is pretty nice even without the stripes. We should
focus on getting our MVP ready for launch instead :)

**Comment by hugolundin** [comment]
Sounds great.
